config-common/src/watcher.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use async_trait::async_trait;
+use notify::Watcher;
+use std::path::PathBuf;
+
+type Api<TConfig, TError> = std::sync::Arc<
+    futures_util::lock::Mutex<dyn UpdateConfig<Config = TConfig, Error = TError> + Send>,
+>;
+
+#[derive(Debug)]
+pub enum ReprovisionTrigger {
+    ConfigurationFileUpdate,
+    Api,
+    Startup,
+}
+
+#[async_trait]
+pub trait UpdateConfig {
+    type Config: serde::de::DeserializeOwned;
+    type Error: std::error::Error;
+
+    async fn update_config(
+        &mut self,
+        new_config: Self::Config,
+        trigger: ReprovisionTrigger,
+    ) -> Result<(), Self::Error>;
+}
+
+pub fn start_watcher<TConfig: 'static, TError: 'static>(
```

Comment:
Lifetime bounds should go along with the other bounds, on lines 35 and 36.

Reply:
Fixed